src/daggy.js

```diff
@@ -169,6 +171,13 @@
     }
   }
 
+  function makeConstructorFromObject(fields) {
+    return function(obj) {
+      var values = fields.map(function(field) { return obj[field]; });
```

Comment:
Let's also add check to throw if a field is not present on `obj`  (note we shouldn't do `obj[key] == null` or something like that as null/undefined is valid value, I guess `hasOwnProperty` should work perfectly)